src/WCSimPrimaryGeneratorAction.cc

```diff
@@ -128,6 +129,12 @@ WCSimPrimaryGeneratorAction::WCSimPrimaryGeneratorAction(
   openangle = 0.;
   wavelength = 435.;
 
+  //Light injector generator variables (LIGen)
+  LIGen = 0;
+  injectorType = "";
+  injectorIdx = "";
+  injectorFilename = "";
+
```

Comment:
I think you're missing initialising of `photonMode = 0`

(It's fine not to initialise if it's given as an option in the mac file. But if it's not, I don't think it is correctly set to the default value. But it's always a complicated flow the messengers, so I may be wrong)